keanu-project/src/main/java/io/improbable/keanu/tensor/validate/NaNFixingTensorValidator.java

```diff
@@ -0,0 +1,23 @@
+package io.improbable.keanu.tensor.validate;
+
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+
+public class NaNFixingTensorValidator implements TensorValidator<Double, DoubleTensor> {
+
+    private final double replacementValue;
+
+    public NaNFixingTensorValidator(double replacementValue) {
+        this.replacementValue = replacementValue;
+    }
+
+    @Override
+    public BooleanTensor check(DoubleTensor tensor) {
+        return tensor.isNaN().not();
```

Comment:
The ND4J tensor does a `.not()` for each `isNan()` and then we do another `.not()` here and then the ChangeValueTo policy gives it another `.not()`. Do we ever need the isNaN to return the individual value? Any way to just replace the NaNs if present?

Reply:
Hmmm - 3 `not`s does seem excessive, doesn't it?

I've replaced `isNaN` with its opposite, `notNaN`. That eliminates the first 2 of the 3 `not`s.

I think we'd probably still want a method named `isNaN` in the API, since it matches the naming convention of `Double#isNaN`.

    default BooleanTensor isNaN() { return notNaN().not(); }

As for the 3rd `not` (in `ChangeValueTo`) - I decided as I worked through it that the semantics made more sense if when `check()` returns `true` it means "everything is fine", however `validate` has to then invert it to create a mask.

Reply:
Also - note that this class doesn't use a `ChangeValueTo` policy, it implements `validate` directly, by calling `tensor.replaceNaN` (which for tensors does the check and replace in one operation, without needing a mask to apply).